sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/KeyPhraseResult.java

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics.models;
+
+import com.azure.core.annotation.Fluent;
+
+import java.util.List;
+
+/**
+ * The KeyPhraseResult model.
+ */
+// TODO (shawn): Should be @Immutable, but will produce spotbug/checkstyle error
+@Fluent
+public final class KeyPhraseResult extends DocumentResult {
+    private List<String> keyPhrases;
+
+    public KeyPhraseResult(String id, TextDocumentStatistics textDocumentStatistics, Error error,
+        List<String> keyPhrases) {
+        super(id, textDocumentStatistics, error);
+        this.keyPhrases = keyPhrases;
+    }
+
+    public List<String> getKeyPhrases() {
+        return keyPhrases;
+    }
+
+    KeyPhraseResult setKeyPhrases(List<String> keyPhrases) {
```

Comment:
Do we need this setter then if the comment suggests this class is immutable.

Reply:
updated to immutable